File: src/cpp/Source/Chunks/Shex/Tokens/OpcodeToken.cpp
#include "PCH.h"
#include "OpcodeToken.h"

using namespace std;
using namespace SlimShader;

const OpcodeHeader& OpcodeToken::GetHeader() const { return _header; }
void OpcodeToken::SetHeader(OpcodeHeader header) { _header = header; }

ostream& SlimShader::operator<<(ostream& out, const OpcodeToken& value)
{
	value.Print(out);
	return out;
}

string OpcodeToken::GetTypeDescription() const
{
	return ToString(_header.OpcodeType);
}

void OpcodeToken::Print(std::ostream& out) const
{
	out << GetTypeDescription();
}

File: src/SlimShader/Chunks/Shex/Tokens/OpcodeToken.cs
namespace SlimShader.Chunks.Shex.Tokens
{
	public abstract class OpcodeToken
	{
		public OpcodeHeader Header { get; internal set; }

		protected string TypeDescription
		{
			get { return Header.OpcodeType.GetDescription(); }
		}

		protected OpcodeToken()
		{
			Header = new OpcodeHeader();
		}

		public override string ToString()
		{
			return TypeDescription;
		}
	}
}